app.py

```diff
@@ -20,15 +22,28 @@ def get_model(self, num_labels: int):
         )
         return model, tokenizer
 
-    def get_dataset_name(self):
-        return "YelpReviewFull", 5
+    def get_dataset(self):
+        data_root_path = os.path.expanduser("~/.cache/torchtext")
+        train_dset = torchtext.datasets.YelpReviewFull(
+            root=data_root_path, split="train"
+        )
+        val_dset = torchtext.datasets.YelpReviewFull(root=data_root_path, split="test")
+        num_labels = 5
+        return train_dset, val_dset, num_labels
+
+    def get_trainer_settings(self):
+        return dict(strategy="deepspeed_stage_3_offload", precision=16)
+
+    def finetune(self):
+        train_dset, val_dset, num_labels = self.get_dataset()
+        module, tokenizer = self.get_model(num_labels)
+        pl_module = TextClassification(model=module, tokenizer=tokenizer)
+        datamodule = TextClassificationData(
+            train_dataset=train_dset, val_dataset=val_dset, tokenizer=tokenizer
+        )
+        trainer = L.Trainer(**self.get_trainer_settings())
 
-    def run(self):
-        super().run()
-        if self.is_main_process:
-            num_stars = self.predict(sample_text) + 1
-            print("Review text:\n", sample_text)
-            print("Predicted rating:", "★" * num_stars)
+        trainer.fit(pl_module, datamodule)
```

Comment:
If we have a method called `finetune()` it might make sense to have the API for this look like `trainer.finetune()` instead of `trainer.fit()` to the end user. `trainer.finetune()` can very much use `.fit()` under the hood. wdyt @edenlightning ?

Reply:
we cannot easily introduce new api to the core trainer as the api there is stable and should not change anymore

Reply:
I wasn't referring to adding another api to the core trainer. I was only referring to exposing the api from the core trainer to the end user as `finetune()` via some abstraction maybe in the TextClf base class?

Reply:
I wouldn't do that since then it's hard to get for them what happens under the hood, which is what they should be able to see. Especially since we are using the raw pytorch lightning trainer.